src/client/common/process/pythonExecutionFactory.ts

```diff
@@ -83,17 +97,43 @@ export class PythonExecutionFactory implements IPythonExecutionFactory {
         });
     }
     public async createActivatedEnvironment(options: ExecutionFactoryCreateWithEnvironmentOptions): Promise<IPythonExecutionService> {
+        const pythonPath = options.interpreter ? options.interpreter.path : this.configService.getSettings(options.resource).pythonPath;
+        const condaExecutionService = await this.createCondaExecutionService(pythonPath, undefined, options.resource);
```

Comment:
If you do the following then you can simplify the signature of `createCondaExecutionService()` (make `processService required and drop `resource`).

```suggestion
        const processService = await this.processServiceFactory.create(resource);
        const condaExecutionService = await this.createCondaExecutionService(pythonPath, processService);
```

Reply:
It would actually add a few more lines than that, because it turns out we also need to call
```ts
procService.on('exec', processLogger.logProcess.bind(processLogger));
this.serviceContainer.get<IDisposableRegistry>(IDisposableRegistry).push(procService);
```
after creating a `processService` object. Then we would have to remember to add these lines wherever `createCondaExecutionService` is called or some mystifying error will happen  at some point 🔮